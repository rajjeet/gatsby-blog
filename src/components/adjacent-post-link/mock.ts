export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
