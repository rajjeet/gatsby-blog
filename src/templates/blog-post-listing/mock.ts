import { mockPosts } from '../../fixtures/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: mockPosts,
    },
  },
  pageContext: {
    numOfPages: 2,
    currentPage: 3,
    tag: 'Sample Tag',
  },
});
