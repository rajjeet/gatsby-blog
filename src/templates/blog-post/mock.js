import { createMockPosts } from '../../utils/testing';

export const makeProps = () => ({
  className: {},
  data: {
    post: createMockPosts[1].node,
  },
});
