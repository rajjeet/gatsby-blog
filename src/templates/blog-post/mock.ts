import { TProps } from './types';
import { mockPost } from '../../fixtures/testing';

export const makeProps = ({ dateModified = null } = {}): TProps => {
  if (dateModified) {
    mockPost.frontmatter.dateModified = dateModified;
  }
  return {
    data: {
      post: { ...mockPost },
    },
  };
};
