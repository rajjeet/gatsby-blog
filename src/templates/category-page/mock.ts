import { createMockPosts } from '../../utils/testing';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    category: 'Self Improvement',
    numOfPages: 2,
    currentPage: 3,
    paginationSlug: '/self-improvement/',
  },
});
