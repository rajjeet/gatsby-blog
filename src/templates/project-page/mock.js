import { createMockPosts, createMockProjects } from '../../utils/testing';

export const makeProps = () => ({
  data: {
    project: createMockProjects[0].node,
    posts: {
      edges: createMockPosts,
    },
  },
});
