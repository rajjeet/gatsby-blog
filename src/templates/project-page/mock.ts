import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';
import { mdxBody } from '../../fixtures/mdx';

export const makeProps = (): TProps => ({
  data: {
    project: {
      frontmatter: {
        title: 'Condo Landing Pages',
        description: 'An automated process for building fast, responsive landing pages for pre-construction condos.',
        techStackTags: [
          {
            type: 'AWS',
            label: 'AWS CloudFormation',
          },
        ],
        links: [
          {
            label: 'Github Code',
            value: 'https://github.com/rajjeet/charismacondos',
          },
        ],
      },
      body: mdxBody,
    },
    posts: {
      edges: mockPosts,
    },
  },
});
