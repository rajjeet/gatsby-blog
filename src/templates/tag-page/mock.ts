import { createMockPosts } from '../../utils/testing';
import { TProps } from './index';

export const makeProps = (): TProps => ({
  data: {
    posts: {
      edges: createMockPosts,
    },
  },
  pageContext: {
    limit: 5,
    skip: 2,
    numOfPages: 2,
    numOfPosts: 10,
    currentPage: 3,
    tag: 'Life Skills',
    paginationSlug: '',
  },
});
