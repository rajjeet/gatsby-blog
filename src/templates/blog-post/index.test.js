import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { StaticQuery } from 'gatsby';
import BlogPost from './index';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import siteMetadata from '../../../gatsby-config';
import { makeProps } from './mock';

afterEach(cleanup);

beforeEach(() => {
  StaticQuery.mockImplementation(({ render: renderQuery }) => renderQuery(
    {
      file: createMockGatsbyImageSharpFluid.file,
      site: siteMetadata,
    },
  ));
});

describe('BlogPost', () => {
  it('should render', () => {
    const { asFragment } = render(<BlogPost {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have a table of contents', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText('Outline')).toBeDefined();
  });

  it('should show the post title', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText('State Management using React Hooks')).toBeDefined();
  });

  it('should show the post date', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText(/02 May, 2019/)).toBeDefined();
  });

  it('should show the time to read', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText(/4 min read/)).toBeDefined();
  });

  it('should show comments', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText(/comments/i)).toBeDefined();
  });

  it('should show the description', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText(/I was learning about the new React Hooks/)).toBeDefined();
  });

  it('should show the blog category', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText('Guide X')).toBeDefined();
  });

  it('should shows the blog tag', () => {
    const { getByText } = render(<BlogPost {...makeProps()} />);
    expect(getByText('React X')).toBeDefined();
    expect(getByText('Angular X')).toBeDefined();
  });

  // it('should shows the blog html', () => {
  //   const { getByText } = render(<BlogPost {...makeProps()} />);
  //   expect(getByText('This is blog content 102')).toBeDefined();
  // });

  // TODO: make mobile tests valid
  // it.skip('should not show the floating button', () => {
  //   const { queryByLabelText } = render(<BlogPost {...makeProps()} />);
  //   console.log(window.innerWidth);
  //   expect(queryByLabelText('Open table of contents')).toBeNull();
  // });

  describe('mobile view', () => {
    beforeEach(() => {
      window.innerWidth = 200;
      window.dispatchEvent(new Event('resize'));
    });

    // TODO: make mobile tests valid
    // it.skip('should not show the table of contents in sidebar', () => {
    //   const { queryByText, debug } = render(<BlogPost {...makeProps()} />);
    //   expect(queryByText('Outline')).toBeNull();
    // });

    it('should show the floating button', () => {
      const { getByLabelText } = render(<BlogPost {...makeProps()} />);
      expect(getByLabelText('Open table of contents')).toBeDefined();
    });

    it('should open the table of contents modal when floating button is clicked', () => {
      const { getByLabelText, getByTestId } = render(<BlogPost {...makeProps()} />);
      fireEvent.click(getByLabelText('Open table of contents'));
      expect(getByTestId('mobile-toc')).toBeDefined();
    });
  });
});
