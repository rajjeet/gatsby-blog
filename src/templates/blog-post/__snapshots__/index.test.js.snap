// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost should render 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat faJHDz"
  >
    <div
      class="sc-bwzfXH bmdiOC"
    >
      <a
        aria-label="See Home"
        href="/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden; width: 3em;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Ortmesh logo"
              src=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;img src="" alt="Ortmesh logo" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </a>
      <a
        href="/"
      >
        <h2>
          Ortmesh
        </h2>
        <p>
          Write code that matters
        </p>
      </a>
      <button
        class="sc-bdVaJa jCUGPA"
      >
        About
      </button>
    </div>
    <div
      class="[object Object] sc-EHOje inCCSA"
    >
      <div
        class="post-summary"
      >
        <h1>
          State Management using React Hooks
        </h1>
        <span>
          02 May, 2019 - 4 min read - 
          <span
            class="disqus-comment-count"
            data-disqus-identifier="5f2933ac-25ed-5ae0-84d7-c5d59fc4448e"
            data-disqus-url="http://ortmesh.com/state-management-react-hooks/"
          >
            Comments
          </span>
        </span>
        <p>
          I was learning about the new React Hooks, when I came across a blog post that claims to build state management, like Redux, using React Hooks. I decided to try it out and found it surprisingly simple and effective.
        </p>
        <div
          class="sc-bxivhb fXNsAS"
        >
          <span
            role="link"
            tabindex="0"
          >
            React X
          </span>
          <span
            role="link"
            tabindex="0"
          >
            Angular X
          </span>
        </div>
      </div>
      <br />
      <div>
        <div
          class="sc-ifAKCX bZyvYn"
        >
          <h4>
            Outline
          </h4>
          <div />
        </div>
        <div>
          <p>
            This is blog content 102
          </p>
        </div>
        <br />
        <div
          id="disqus_thread"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
